rest/api_test.go

```diff
@@ -5313,3 +5314,139 @@ func TestAttachmentContentType(t *testing.T) {
 		assert.Equal(t, "", contentDisposition)
 	}
 }
+
+func TestDeleteNonExistentDoc(t *testing.T) {
+	rt := NewRestTester(t, nil)
+	defer rt.Close()
+
+	response := rt.SendAdminRequest("DELETE", "/db/fake", "")
+	assertStatus(t, response, http.StatusOK)
+
+	response = rt.SendAdminRequest("GET", "/db/fake", "")
+	assertStatus(t, response, http.StatusNotFound)
+
+	var body map[string]interface{}
+	_, err := rt.GetDatabase().Bucket.Get("fake", &body)
+
+	if base.TestUseXattrs() {
+		assert.Error(t, err)
```

Comment:
We should validate the specific error we're expecting here, to catch regressions if other errors start appearing in future.